### PR TITLE
Refacto/profile

### PR DESCRIPTION
**Pull request by kevin-verschaeve**

### CODE REVIEW

File: front/services/address/AddressForm.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ export default class AddressForm {
         addClass(select, 'form__field');
         addClass(select, 'em-form__field');
 
-        insertAfter(this._city, select);
+        insertAfter(this._cityName, select);
```
Comment (ottaviano) [inline front/services/address/AddressForm.js:126]: why do we change this?

File: front/style/layout/_avatar.scss [hunk 2]
```diff
@@ -43,6 +43,11 @@
     }
 
     &--style-02 {
+        background-color: $pink;
```
Comment (ottaviano) [inline front/style/layout/_avatar.scss:46]: @enzozccr, do we really want to change this? (it changes the style of avatars on CoTerr pages)

File: front/style/layout/_utilities.scss [hunk 3]
```diff
@@ -11,3 +11,7 @@
 .visually-hidden {
     display: none;
 }
+
+.invisible {
```
Comment (ottaviano) [inline front/style/layout/_utilities.scss:15]: @enzozccr, it seems to be unused?

File: front/style/pages/_adherent-profile.scss [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        //     scrollbar-width: none;
+        // }
+
+        @media (min-width: $tablet) {
+            display: block;
+        }
+    }
+
+    &__status {
+
```
Comment (ottaviano) [inline front/style/pages/_adherent-profile.scss:35]: maybe remove this empty block?

File: public/apps/projets-citoyens/yarn.lock [hunk 5]
```diff
@@ -5499,11 +5499,16 @@ lodash.uniq@^4.5.0:
   resolved "https://registry.yarnpkg.com/lodash.uniq/-/lodash.uniq-4.5.0.tgz#d0225373aeb652adc1bc82e4945339a842754773"
   integrity sha1-0CJTc662Uq3BvILklFM5qEJ1R3M=
 
-"lodash@>=3.5 <5", lodash@^4.0.0, lodash@^4.14.0, lodash@^4.15.0, lodash@^4.17.19, lodash@^4.17.2, lodash@^4.17.3, lodash@^4.17.4, lodash@^4.17.5, lodash@^4.2.0, lodash@^4.2.1, lodash@^4.3.0, lodash@~4.17.10, lodash@~4.17.4:
+"lodash@>=3.5 <5", lodash@^4.0.0, lodash@^4.14.0, lodash@^4.15.0, lodash@^4.17.2, lodash@^4.17.3, lodash@^4.17.4, lodash@^4.17.5, lodash@^4.2.0, lodash@^4.2.1, lodash@^4.3.0, lodash@~4.17.10, lodash@~4.17.4:
```
Comment (ottaviano) [inline public/apps/projets-citoyens/yarn.lock:5502]: @kevin-verschaeve, it seems to be an error on conflict resolution and should not be part of this PR

File: front/style/layout/_nav.scss [hunk 6]
```diff
@@ -286,3 +287,70 @@
...
+.em-nav-mobile {
+    display: none;
+    width: 100%;
+    background: $white;
+
+    &__adherent-profile {
+        background: $gray--lighter;
+    }
+
+    &__sitemap {
```
Comment (ottaviano) [inline front/style/layout/_nav.scss:332]: could we remove this empty block?

File: templates/adherent_profile/activity/_committees.html.twig [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            <article class="activity-card b__nudge--bottom">
+                <div class="activity-card__type activity-card__type--committee">Comité</div>
+                <h2 class="activity-card__title font-roboto text--medium b__nudge--bottom-small text--breakword text--bold l__row">{{ committee.name }}</h2>
+                <div class="l__row font-roboto text--gray b__nudge--bottom-small">
+                    <div class="l__row b__nudge--right-small">
+                        <img src="/images/icons/activity/place.svg" alt="Ville" class="b__nudge--right-nano">
+                        <span>{{ committee.getCityName }}</span>
+                    </div>
+                    <div class="l__row">
+                        <img src="/images/icons/activity/members.svg" alt="Adhérents" class="b__nudge--right-nano">
```
Comment (ottaviano) [inline templates/adherent_profile/activity/_committees.html.twig:40]: `src="{{ asset('/images/icons/activity/members.svg') }}"`

File: templates/adherent_profile/activity/_events.html.twig [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+                            <div class="text--body text--gray b__nudge--bottom-small">Organisé par un ancien adhérent à <span class="text--black">{{ event.cityName }}</span></div>
+                        {% endif %}
+                    {% elseif event.committee %}
+                        <div class="text--body text--gray b__nudge--bottom-small">Organisé par le comité <span class="text--black">{{ event.committee.name }}</span> à <span class="text--black">{{ event.cityName }}</span></div>
+                    {% elseif event.organizer %}
+                        <div class="text--body text--gray b__nudge--bottom-small">Organisé par <span class="text--black">{{ is_granted('IS_AUTHENTICATED_FULLY') ? event.organizer.fullName : event.organizer.partialName }}</span> à <span class="text--black">{{ event.cityName }}</span></div>
+                    {% endif %}
+                    <div class="l__row font-roboto text--gray b__nudge--bottom-small">
+                        <div class="l__row b__nudge--bottom-small">
+                            <img src="/images/icons/activity/date.svg" alt="Ville" class="b__nudge--right-nano">
```
Comment (ottaviano) [inline templates/adherent_profile/activity/_events.html.twig:37]: `src="{{ asset(...) }}"`

File: templates/adherent_profile/activity/_events.html.twig [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
```
Comment (ottaviano) [inline templates/adherent_profile/activity/_events.html.twig:94]: `{{ asset(...) }}`

File: front/style/layout/_avatar.scss (hunk 2 above)
Comment (enzozccr) [reply front/style/layout/_avatar.scss:46]: ah, need to create a style-03 instead

File: front/style/pages/_adherent-profile.scss [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+        padding-bottom: 30px;
+        flex: 0 0 305px;
+        display: none;
+        position: sticky;
+        top: 0;
+        align-self: flex-start;
+        max-height: 100vh;
+        overflow-y: auto;
+
+        // Hide nav scroll ?
```
Comment (illusionOfParadise) [inline front/style/pages/_adherent-profile.scss:22]: Still need to be commented ?

File: front/services/address/AddressForm.js (hunk 1 above)
Comment (kevin-verschaeve) [reply front/services/address/AddressForm.js:126]: If need to change this because the field is not rendered at the good place. Can't found an other way to do it

File: src/Controller/EnMarche/AdherentProfile/FunnelController.php [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+use App\Form\AdherentFunnelGeneralType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+class FunnelController extends AbstractController
+{
+    /**
+     * @Route("/funnel/general", name="app_funnel_general")
+     */
+    public function generalAction(Request $request, AdherentProfileHandler $handler)
```
Comment (illusionOfParadise) [inline src/Controller/EnMarche/AdherentProfile/FunnelController.php:17]: Return type missed.

File: src/Controller/EnMarche/AdherentProfile/FunnelController.php [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+        return $this->render('adherent_profile/funnel/general.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/funnel/certification", name="app_funnel_certification")
+     */
+    public function certificationAction()
```
Comment (illusionOfParadise) [inline src/Controller/EnMarche/AdherentProfile/FunnelController.php:39]: Return type missed.

File: src/AdherentProfile/AdherentProfile.php [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+     *
+     * @Assert\NotBlank(message="adherent.birthdate.not_blank")
+     * @Assert\Range(max="-15 years", maxMessage="adherent.birthdate.minimum_required_age")
+     */
+    private $birthdate;
+
+    /**
+     * @var bool
+     */
+    private $certified = false;
```
[the remark this answers is not in the document]
Comment (kevin-verschaeve) [reply src/AdherentProfile/AdherentProfile.php]: done

File: src/Controller/EnMarche/AdherentProfile/FunnelController.php (hunk 11 above)
Comment (kevin-verschaeve) [reply src/Controller/EnMarche/AdherentProfile/FunnelController.php]: fixed

File: src/Entity/Adherent.php [hunk 14]
```diff
@@ -536,6 +568,13 @@ class Adherent implements UserInterface, UserEntityInterface, GeoPointInterface,
      */
     private $receivedDelegatedAccesses;
 
+    /**
+     * @var AdherentCommitment
+     *
+     * @ORM\OneToOne(targetEntity="App\Entity\AdherentCommitment", mappedBy="adherent", cascade={"all"}, fetch="EXTRA_LAZY")
```
Comment (illusionOfParadise) [inline src/Entity/Adherent.php:574]: `EXTRA_LAZY` has no sense here.
Inverse sides of one-to-one associations can not be lazy, technically. There is no foreign key on the inverse side, hence it is impossible to decide whether to proxy it or not.

File: src/Entity/Adherent.php [hunk 15]
```diff
@@ -1058,6 +1097,28 @@ public function setInterests(array $interests): void
         $this->interests = $interests;
     }
 
+    public function updateProfile(AdherentProfile $adherentProfile, PostAddress $postAddress)
```
Comment (illusionOfParadise) [inline src/Entity/Adherent.php:1100]: Return type is missing.

File: src/Entity/Adherent.php [hunk 16]
```diff
@@ -2195,45 +2256,73 @@ public function isLre(): bool
...
-        if ($this->isReferent()) {
-            $tags = $this->getManagedArea()->getReferentTagCodes();
-        }
+        return $this->facebookPageUrl;
+    }
 
-        foreach ($this->getReceivedDelegatedAccessOfType('referent') as $delegatedAccess) {
-            $tags = \array_merge($tags, $delegatedAccess->getDelegator()->getManagedArea()->getReferentTagCodes());
-        }
+    public function setFacebookPageUrl($facebookPageUrl): void
```
Comment (illusionOfParadise) [inline src/Entity/Adherent.php:2264]: It should be possible to set `null` to `facebookPageUrl`.
Same for other urls' setters.

File: src/Entity/Adherent.php (hunk 14 above)
Comment (illusionOfParadise) [inline src/Entity/Adherent.php:574]: `EXTRA_LAZY` has no sense here.
Inverse sides of one-to-one associations can not be lazy, technically. There is no foreign key on the inverse side, hence it is impossible to decide whether to proxy it or not.

File: src/Entity/AdherentCommitment.php [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        $this->skills = $skills;
+    }
+
+    public function getAvailability(): ?string
+    {
+        return $this->availability;
+    }
+
+    public function setAvailability(string $availability): void
```
Comment (illusionOfParadise) [inline src/Entity/AdherentCommitment.php:153]: Can be nullable.

File: templates/adherent_profile/_layout.html.twig [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+                                </div>
+                                <div class="adhesion-date">Inscrit{{ app.user.female ? "e" }} en {{ app.user.registeredAt|localizeddate('none', 'none', 'fr_FR', null, 'MMMM yyyy') }}</div>
+                            </div>
+                        </div>
+                        <ul class="adherent-profile__roles b__nudge--bottom-large">
+                            {% for role in get_adherent_role_labels(app.user) %}
+                                <li>{{ role }}</li>
+                            {% endfor %}
+                        </ul>
+{#                        <div class="adherent-profile__completion b__nudge--bottom-50">#}
```
Comment (illusionOfParadise) [inline templates/adherent_profile/_layout.html.twig:79]: Still need to be commented ?

File: templates/adherent_profile/activity/_adi.html.twig [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+        <li{% if type == 'adi_contributions' %} class="active"{% endif %}><a href="#" class="tabs-nav__item js--tabs" data-target="#adi-my-contributions">Mes contributions</a></li>
+        <li{% if type == 'adi_votes' %} class="active"{% endif %}><a href="#" class="tabs-nav__item js--tabs" data-target="#adi-my-votes">Mes votes</a></li>
+    </ul>
+</nav>
+
+<div id="adi-my-proposals" class="js--tabs-content{{ type is not null and isAdiSection and type != 'adi_proposals' ? ' visually-hidden' }}">
+    <div class="l__col">
+        {% if adi_proposals.totalItems < 1 %}
+            <div class="space--40-0 text--center">
+                <p class="font-roboto text--default-large text--bold b__nudge--bottom">Contribuez à l'Atelier des Idées</p>
```
Comment (illusionOfParadise) [inline templates/adherent_profile/activity/_adi.html.twig:16]:
```suggestion
                <p class="font-roboto text--default-large text--bold b__nudge--bottom">Contribuez à l'Atelier des idées</p>
```

File: templates/adherent_profile/informations/elections.html.twig [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+        <div class="text--body text--black text--bold b__nudge--none b__nudge--bottom-large">
+            Vos mandats en cours
+        </div>
+        <div class="data-display">
+            <div class="data-display__item">
+                <div class="label">Périmètre géographique</div>
+                <div class="value">1ère circonscription du Val-de-Marne</div>
+            </div>
+            <div class="data-display__item">
+                <div class="label">Label</div>
```
Comment (illusionOfParadise) [inline templates/adherent_profile/informations/elections.html.twig:28]: Is a content of this page is temporary?

File: templates/certification_request/home.html.twig [hunk 21]
```diff
@@ -1,54 +1,59 @@
...
+                Vous pouvez <a class="link--blue--dark link--no-decor" href="{{ path('app_certification_request_form') }}">réessayer</a>.</p>
 
-            {% if app.user.birthdate %}
-                <a href="{{ path('app_certification_request_form') }}" class="btn btn--blue b__nudge--top">Certifier mon profil</a>
             {% else %}
-                <div class="b__nudge--top-40 alert alert--error">Malheureusement, nous ne connaissons pas votre date de naissance et nous en avons besoin pour certifier votre identité. Mettez à jour votre profil avant de continuer.</div>
-                <p><a href="{{ path('app_user_edit') }}" class="btn btn--blue b__nudge--top">METTRE À JOUR MON PROFIL</a></p>
+                <img src="{{ asset('/images/icons/certification/not-certified.svg') }}" alt="certif-img"/>
+                <div class="title b__nudge--bottom">Votre profil n'est pas encore certifié.</div>
+                <p>Certifiez le avec une pièce d'identité, ça ne prend qu'une minute.</p>
```
Comment (illusionOfParadise) [inline templates/certification_request/home.html.twig:45]:
```suggestion
                <p>Certifiez-le avec une pièce d'identité, ça ne prend qu'une minute.</p>
```

File: templates/components/_nav_mobile.html.twig [hunk 22]
```diff
@@ -10,314 +10,203 @@
...
+                    {% if has_role_adherent %}
+                        <div class="adherent-profile__id l__row b__nudge--bottom">
+                            <div class="name b__nudge--right-nano">{{ app.user.partialName }}</div>
+                        </div>
+                        <ul class="adherent-profile__roles b__nudge--bottom">
+                            {% for role in get_adherent_role_labels(app.user) %}
+                                <li>{{ role }}</li>
+                            {% endfor %}
                         </ul>
+{#                        <div class="adherent-profile__completion b__nudge--bottom">#}
```
Comment (illusionOfParadise) [inline templates/components/_nav_mobile.html.twig:25]: Still should be commented?

File: src/Controller/EnMarche/AdherentProfile/CommitmentController.php [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class CommitmentController extends AbstractController
+{
+    /**
+     * @Route("/parametres/engagement", name="app_adherent_commitment")
+     */
+    public function indexAction(Request $request, UserInterface $adherent, EntityManagerInterface $entityManager)
```
Comment (ottaviano) [inline src/Controller/EnMarche/AdherentProfile/CommitmentController.php:18]: typehint

File: src/Controller/EnMarche/AdherentProfile/FunnelController.php [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+use App\Controller\EnMarche\VotingPlatform\AbstractController;
+use App\Form\AdherentFunnelGeneralType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class FunnelController extends AbstractController
+{
+    /**
+     * @Route("/funnel/general", name="app_funnel_general")
```
Comment (ottaviano) [inline src/Controller/EnMarche/AdherentProfile/FunnelController.php:16]: `method={...}`

File: tests/Controller/EnMarche/AdherentControllerTest.php [hunk 25]
```diff
@@ -63,38 +62,44 @@ public function testAuthenticatedAdherentCanSeeHisUpcomingAndPastEvents(): void
...
+        self::assertEquals('http://test.enmarche.code/parametres/mes-activites?page=2&type=past_events#events', $crawler->getUri());
+        $titles = $crawler->filter('#past-events h2');
+        $this->assertSame(2, $crawler->filter('#past-events article')->count());
+        $this->assertSame('Grand Meeting de Paris', trim($titles->first()->text()));
         $this->assertSame('Grand Meeting de Marseille', trim($titles->last()->text()));
     }
 
     /**
      * @dataProvider provideProfilePage
+     * @group
```
Comment (illusionOfParadise) [inline tests/Controller/EnMarche/AdherentControllerTest.php:102]: Can be removed.

File: src/Form/AdherentProfileType.php [hunk 26]
```diff
@@ -0,0 +1,112 @@
...
+                'required' => false,
+            ])
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefined('disabled_form');
+        $resolver->setAllowedTypes('disabled_form', 'bool');
+        $resolver->setDefault('disabled', false);
```
Comment (ottaviano) [inline src/Form/AdherentProfileType.php:106]: `disabled_form`

File: src/Repository/CitizenProjectMembershipRepository.php [hunk 27]
```diff
@@ -286,6 +289,42 @@ public function findCitizenProjectMemberships(CitizenProject $citizenProject): C
...
+        ;
+
+        return $this->configurePaginator($queryBuilder, $page, $limit);
+    }
+
+    public function findActivityCitizenProjectMembershipsJoined(Adherent $adherent, int $page = 1, int $limit = 5)
+    {
+        $queryBuilder = $this
+            ->createQueryBuilder('cpm')
+            ->innerJoin(CitizenProjectMembership::class, 'cpm2', 'WITH', 'cpm.citizenProject = cpm2.citizenProject')
```
Comment (ottaviano) [inline src/Repository/CitizenProjectMembershipRepository.php:314]: `'WITH'` => `Join::WITH`

File: src/Repository/CommitteeMembershipRepository.php [hunk 28]
```diff
@@ -78,6 +79,25 @@ public function superviseCommittee(Adherent $adherent, Committee $committee = nu
         return (int) $qb->getQuery()->getSingleScalarResult() >= 1;
     }
 
+    public function findActivityMemberships(Adherent $adherent, int $page = 1, int $limit = 5)
```
Comment (ottaviano) [inline src/Repository/CommitteeMembershipRepository.php:82]: typehint

File: src/Repository/EventRegistrationRepository.php [hunk 29]
```diff
@@ -104,6 +105,19 @@ public function findUpcomingAdherentRegistrations(string $adherentUuid): array
         ;
     }
 
+    public function findActivityUpcomingAdherentRegistrations(string $adherentUuid, int $page = 1, int $limit = 5)
+    {
+        self::validUuid($adherentUuid);
```
Comment (ottaviano) [inline src/Repository/EventRegistrationRepository.php:110]: why do not pass Adherent instance like for others?

File: src/Repository/EventRegistrationRepository.php [hunk 30]
```diff
@@ -130,6 +144,20 @@ public function findPastAdherentRegistrations(string $adherentUuid): array
         ;
     }
 
+    public function findActivityPastAdherentRegistrations(string $adherentUuid, int $page = 1, int $limit = 5)
+    {
+        self::validUuid($adherentUuid);
```
Comment (ottaviano) [inline src/Repository/EventRegistrationRepository.php:149]: same question here

File: src/Repository/VoteRepository.php [hunk 31]
```diff
@@ -2,16 +2,33 @@
...
 {
     use AuthorTrait;
+    use PaginatorTrait;
 
     public function __construct(RegistryInterface $registry)
     {
         parent::__construct($registry, Vote::class);
     }
+
+    public function getIdeasVotesFromAdherent(Adherent $adherent, int $page = 1, int $limit = 5): PaginatorInterface
```
[the remark this answers is not in the document]
Comment (kevin-verschaeve) [reply src/Repository/VoteRepository.php:21]: No it's not :smile:

File: templates/adherent_profile/_funnel_layout.html.twig [hunk 32]
```diff
@@ -0,0 +1,66 @@
...
+                        <ul class="adherent-profile__roles b__nudge--bottom-large">
+                            {% for role in get_adherent_role_labels(app.user) %}
+                                <li>{{ role }}</li>
+                            {% endfor %}
+                        </ul>
+                        <div class="adherent-profile__completion b__nudge--bottom-50">
+                            <div class="text--body text--small b__nudge--bottom-small">
+                                <!-- If is not completed -->
+                                Complétion du profil : 50%
+                                <!-- If is completed TODO
```
[the remark this answers is not in the document]
Comment (kevin-verschaeve) [reply templates/adherent_profile/_funnel_layout.html.twig:32]: yes, this will be done in an other pr

File: templates/adherent_profile/_layout.html.twig (hunk 18 above)
Comment (kevin-verschaeve) [reply templates/adherent_profile/_layout.html.twig:79]: yes, @enzozccr did the integration, but the feature will be done in an other pr, so I leave it here to not loose it

File: templates/adherent_profile/informations/elections.html.twig (hunk 20 above)
Comment (kevin-verschaeve) [reply templates/adherent_profile/informations/elections.html.twig:28]: this page is integrated by Enzo, but the feature is not ready

File: templates/certification_request/home.html.twig (hunk 21 above)
Comment (kevin-verschaeve) [reply templates/certification_request/home.html.twig:45]: hmm I'm not sure about that

File: src/Repository/EventRegistrationRepository.php (hunk 29 above)
Comment (kevin-verschaeve) [reply src/Repository/EventRegistrationRepository.php:110]: I based this method on this one https://github.com/EnMarche/en-marche.fr/pull/4640/files/54859de557df570e5f1960f43241bdd8b083697c#diff-39f09d2ecd7d2f1787e990441abd3a7fL86
just above

File: templates/certification_request/home.html.twig (hunk 21 above)
Comment (illusionOfParadise) [reply templates/certification_request/home.html.twig:47]: I'm sure 😄: http://www.visezjuste.uottawa.ca/pages/orthographe/pronoms_personnels.html

Comment (kevin-verschaeve) [reply templates/certification_request/home.html.twig:47]: haha ok, thanks !
